### PR TITLE
test cases added

### LINKED ISSUE #113: Budget Tracker Improvements
**Create Speerate PR for UI and Techincal Fix**

## UI
- [x] If there is no list available for Select, then show `No (context) available`, and it should be disabled.
- [x] Settings => Donors :1. Skeleton on loading.

## Technical
- [ ] Replaacce `any` with proper type. Keeps types in sperate file.
- [x] Remaining Test Cases.
- [ ] Added Proper description if any component is being used in multiple places.  Things that description should include are:
  1. What is component about.  
  2. An Example of input / output if possible.
  3. Any Special Case that is left out or is being handled?

### PR DESCRIPTION
**Pull request by shubhamsingh3211**
## Description

Test cases added for the following
1)Budget target
2)Budget line item
3)Donor

Fixes #113 partially fixes the issue mentioned

### CODE REVIEW

File: src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx [hunk 1]
```diff
@@ -208,25 +208,30 @@ beforeEach(() => {
 	});
 });
 
-const inputIds = budgetLineItemInputFields;
+const budgetLineItemInputIds = budgetLineItemInputFields;
 
 describe("Budget Line Item Dialog tests", () => {
```
Comment (ShadabFaiz) [inline src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx:213]: What functionality are we exactly testing here?

File: src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx [hunk 2]
```diff
@@ -208,25 +208,30 @@ beforeEach(() => {
...
-		// 	}
-		// 	await act(async () => {
-		// 		let saveButton = await dialog.getByTestId("createSaveButton");
-		// 		expect(saveButton).toBeEnabled();
-		// 		fireEvent.click(saveButton);
-		// 		await wait();
-		// 	});
-		// 	await new Promise((resolve) => setTimeout(resolve, 1000));
-		// 	expect(creationOccured).toBe(true);
+		await new Promise((resolve) => setTimeout(resolve, 1000));
```
Comment (ShadabFaiz) [inline src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx:215]: What do we need to wait for 1s ? waiting for dom update / data loading ?

File: src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx [hunk 3]
```diff
@@ -208,25 +208,30 @@ beforeEach(() => {
...
+			await expect(budgetLineItemFieldName.value).toBe(budgetLineItemValue);
+		}
+		await act(async () => {
+			let budgetLineItemSaveButton = await dialog.getByTestId("createSaveButton");
+			expect(budgetLineItemSaveButton).toBeEnabled();
+			fireEvent.click(budgetLineItemSaveButton);
+			await wait();
+		});
+		await new Promise((resolve) => setTimeout(resolve, 1000));
+		expect(creationOccured).toBe(true);
```
Comment (ShadabFaiz) [inline src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx:235]: Multiple Expect ? Are we testing multiple functionality in same test suites?

File: src/components/Budget/BudgetTarget/__test__/BudgetTarget.test.tsx [hunk 4]
```diff
@@ -111,49 +118,48 @@ const mocks = [
...
+				addTypename: false,
+			}
+		);
+	});
+});
 
-const inputIds = BudgetTargetinputFields;
+const budgetTargetInputIds = BudgetTargetinputFields;
 
 describe("Budget Target Dialog tests", () => {
```
Comment (ShadabFaiz) [inline src/components/Budget/BudgetTarget/__test__/BudgetTarget.test.tsx:145]: Again..what funtionality are we testing here?

File: src/components/Table/Budget/BudgetLineItemTable/__test__/BudgetLineitemTable.test.tsx [hunk 5]
```diff
@@ -158,7 +158,7 @@ beforeEach(() => {
 
 describe("Budget Line Item Table tests", () => {
 	test("renders correctly", async () => {
-		// await waitForElement(() => table.getByText(/500/i));
-		// await waitForElement(() => table.getByText(/note 1/i));
+		await waitForElement(() => table.getByText(/500/i));
```
Comment (ShadabFaiz) [inline src/components/Table/Budget/BudgetLineItemTable/__test__/BudgetLineitemTable.test.tsx:161]: what is /500/i?

**Review by ShadabFaiz — COMMENTED** [review]
There are multiple place where we are using hard code values to get element. Instead replace them with
enum / variable from the mock data that is being applied into them

File: src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx (hunk 1 above)
Comment (shubhamsingh3211) [reply src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx:213]: I have mentioned the names of the input field and the key to access their values from  an object called intialFormValue in a seprate file. Then i loop through all the input fields and insert the value and check if the values are modified and the input field has the correct value, then I check if the submit button got enabled, then I check if the request got submitted. All of these test are releated to each other,

File: src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx (hunk 3 above)
Comment (shubhamsingh3211) [reply src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx:235]: yes all of the tests are related to each other, that is why nultiple expect.

File: src/components/Table/Budget/BudgetLineItemTable/__test__/BudgetLineitemTable.test.tsx (hunk 5 above)
Comment (shubhamsingh3211) [reply src/components/Table/Budget/BudgetLineItemTable/__test__/BudgetLineitemTable.test.tsx:161]: This data is mentioned in the mock file and I am checking if it is rendered in the table.

File: src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx (hunk 1 above)
Comment (shubhamsingh3211) [reply src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx:213]: I have mentioned the id of the input element in a seprate file and the mocks in seprate file, and initial values of a form is mentioned seprately in every test file.

File: src/components/Budget/BudgetTarget/__test__/BudgetTarget.test.tsx (hunk 4 above)
Comment (shubhamsingh3211) [reply src/components/Budget/BudgetTarget/__test__/BudgetTarget.test.tsx:145]: I have mentioned the names of the input field and the key to access their values from an object called intialFormValue in a seprate file. Then i loop through all the input fields and insert the value and check if the values are modified and the input field has the correct value, then I check if the submit button got enabled, then I check if the request got submitted. All of these test are releated to each other,

File: src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx (hunk 2 above)
Comment (shubhamsingh3211) [reply src/components/Budget/BudgetLineitem/__test__/BudgetLineItem.test.tsx]: I have removed this

**Review by ShadabFaiz — APPROVED** [review]

**Pull request merged**